src/UnitTests/GitHub.App/Services/PullRequestServiceTests.cs

```diff
@@ -79,69 +124,66 @@ public class TheExtractDiffFilesMethod
             var headFileContent = null as string;
             var fileName = "fileName";
             var baseSha = "baseSha";
-            var headSha = null as string;
+            var headSha = "headSha";
             var baseRef = new GitReferenceModel("ref", "label", baseSha, "uri");
-            var headFileSha = "headFileSha";
             var checkedOut = true;
             var workingFile = Path.Combine(repoDir, fileName);
 
-            var files = await ExtractDiffFiles(baseFileContent, headFileContent, fileName, baseSha, headSha, headFileSha, checkedOut, repoDir);
+            var files = await ExtractDiffFiles(baseSha, baseFileContent, headSha, headFileContent, baseSha, baseFileContent,
+                fileName, checkedOut, repoDir);
 
             Assert.Equal(baseFileContent, File.ReadAllText(files.Item1));
             Assert.Equal(workingFile, files.Item2);
         }
 
         [Fact]
-        public async Task HeadBranchNotAvailable_GetFileFromModelService()
+        public async Task HeadBranchNotAvailable_ThrowsFileNotFoundException()
         {
             var baseFileContent = "baseFileContent";
             var headFileContent = new FileNotFoundException();
-            var headFile = "headFile";
             var fileName = "fileName";
             var baseSha = "baseSha";
             var headSha = "headSha";
             var baseRef = new GitReferenceModel("ref", "label", baseSha, "uri");
             var headRef = new GitReferenceModel("ref", "label", headSha, "uri");
-            var headFileSha = "headFileSha";
             var checkedOut = false;
 
-            var files = await ExtractDiffFiles(baseFileContent, headFileContent, fileName, baseSha, headSha, headFileSha, checkedOut, modelServiceHeadFile: headFile);
+            FileNotFoundException ex = null;
+            try
+            {
+                // Can't use Assert.Throws because ExtractDiffFiles is async.
+                await ExtractDiffFiles(baseSha, baseFileContent, headSha, headFileContent, baseSha, baseFileContent,
```

Comment:
It looks like you can indeed. I hadn't noticed `Assert.ThrowsAsync` before. Thanks for the tip!